### DISCUSSION TITLE
extraout variables

### DISCUSSION THREAD

**Discussion by KingRognar** [NationalSecurityAgency/ghidra#7323] [Q&A]
Hello there!
i'm new to this tool and to decompiling in general, what does "extraout_" variables mean and how can i remove them?

**Comment by KingRognar** [comment]
i googled it for some time and found out only that it's because of function return is not properly defined, i have tried to change it to what i thought should work but it still didn't work for me

**Comment by DualTachyon** [comment]
It is hard to explain in a short comment, specially you provide no details on your binary / CPU / etc, but it is possible and likely that some of your functions are using a different register convention than the one you have loaded (or that you loaded the wrong register convention).

Sometimes it can be fixed with edits to the function signature, sometimes you need to create a new register convention (e.g. a copy of __stdcall ) and update which registers are used on the way in, out and which are killed.

**Comment by KingRognar** [comment]
i'm working with Sega Genesis game (ROM) and it uses M68000 CPU
i wanted to share images but i don't really know what will be usefull to you

**Comment by KingRognar** [comment]
```assembly
bsr.w      UpdateCharacterColliders
move.w     collCount,(offset NumberOfPlayerColliders,ram_pntr)
lea        (0x164c,ram_pntr),A0
move.w     (0x1a3c,ram_pntr),collCount
subq.w     #0x1,collLeft
                             LAB_00009a6a
movea.l    (A0),A4
move.w     (0x4,A0),local_D0w_88
                             LAB_00009a70
move.w     (0x6,A0),t_upCollBrdr
move.w     (0x8,A0),uVar9
move.w     (0xa,A0),uVar10
move.w     (0x8,A4),projtIsActive
tst.b      (0x40,A4)
bpl.b      Hostile_Proj
```
```C
    NumberOfPlayerColliders = UpdateCharacterColliders(2,&Coll_P1,&Char_CollDimensions,&Data_P1);
    projCollData_pntr = &Coll_Proj_1;
    projCounter = NumberOfProjectileColliders - 1;
    uVar1 = extraout_D0;
    do {
      proj_pntr = projCollData_pntr->entityAdress;
      extraout_D1w = (undefined2)((uint)uVar1 >> 0x10);
      uVar1 = CONCAT22(extraout_D1w,projCollData_pntr->leftBorder);
      uVar6 = (uint)projCollData_pntr->upperBorder;
      uVar9 = (uint)projCollData_pntr->rightBorder;
      uVar10 = (uint)projCollData_pntr->bottomBorder;
      projtIsActive = (uint)proj_pntr->isActive?;
      if ((char)proj_pntr->isHostile? < uVar7) {
```
![image](https://github.com/user-attachments/assets/583fe396-ae38-4081-81c6-6c0fcc9d6d50)
i assume that 9a6c move.w should simply overwrite 2 bytes and doesn't need to know what was there before
